manager/volume.go

```diff
@@ -216,65 +207,11 @@ func (m *VolumeManager) Create(name string, spec *longhorn.VolumeSpec, recurring
 	// make sure it's multiples of 4096
 	size = util.RoundUpSize(size)
 
-	if spec.NumberOfReplicas == 0 {
-		spec.NumberOfReplicas, err = m.getDefaultReplicaCount()
-		if err != nil {
-			return nil, errors.Wrap(err, "BUG: cannot get valid number for setting default replica count")
-		}
-		logrus.Infof("Use the default number of replicas %v", spec.NumberOfReplicas)
-	}
-
-	if string(spec.ReplicaAutoBalance) == "" {
-		spec.ReplicaAutoBalance = longhorn.ReplicaAutoBalanceIgnored
-		logrus.Infof("Use the %v to inherit global replicas auto-balance setting", spec.ReplicaAutoBalance)
-	}
-	if err := types.ValidateReplicaAutoBalance(spec.ReplicaAutoBalance); err != nil {
-		return nil, errors.Wrapf(err, "cannot create volume with replica auto-balance %v", spec.ReplicaAutoBalance)
-	}
-
-	if string(spec.DataLocality) == "" {
-		defaultDataLocality, err := m.GetSettingValueExisted(types.SettingNameDefaultDataLocality)
-		if err != nil {
-			return nil, errors.Wrapf(err, "cannot get valid mode for setting default data locality for volume: %v", name)
-		}
-		spec.DataLocality = longhorn.DataLocality(defaultDataLocality)
-	}
-	if err := types.ValidateDataLocality(spec.DataLocality); err != nil {
-		return nil, errors.Wrapf(err, "cannot create volume with data locality %v", spec.DataLocality)
-	}
-
-	if string(spec.AccessMode) == "" {
-		spec.AccessMode = longhorn.AccessModeReadWriteOnce
-	}
-
-	if spec.Migratable && spec.AccessMode != longhorn.AccessModeReadWriteMany {
-		return nil, fmt.Errorf("migratable volumes are only supported in ReadWriteMany (rwx) access mode")
-	}
-
 	defaultEngineImage, err := m.GetSettingValueExisted(types.SettingNameDefaultEngineImage)
 	if defaultEngineImage == "" {
 		return nil, fmt.Errorf("BUG: Invalid empty Setting.EngineImage")
 	}
```

Comment:
Redundant. This is already moved into the validator.

Reply:
Oh, sorry. It's not in the validator. We may need to move this to the validator.

Reply:
Ah, thank you. I forgot this part.